src/simplify-type.js

```diff
@@ -44,15 +60,15 @@ function possibleTypes(interfaceType) {
   return interfaceType.possibleTypes.map((t) => t.name);
 }
 
-export default function simplifyType(type) {
+export default function simplifyType(type, whiteList = false) {
   const simplifiedType = {
     name: type.name,
     kind: type.kind
   };
 
   switch (type.kind) {
     case 'OBJECT':
-      simplifiedType.fieldBaseTypes = fieldBaseTypes(type.fields);
+      simplifiedType.fieldBaseTypes = fieldBaseTypes(type.fields, whiteList);
```

Comment:
I mean that the `fieldBaseTypes` object is set here which isn't recursive? It could just be reduced after the fact so limit the whitelist concept to this function.